client/state/partner-portal/licenses/hooks/use-products-query.ts

```diff
@@ -49,11 +50,14 @@ function queryProducts(): Promise< APIProductFamily[] > {
 export default function useProductsQuery(): UseQueryResult< APIProductFamilyProduct[], unknown > {
 	const translate = useTranslate();
 	const dispatch = useDispatch();
+	const isPartnerOAuthTokenLoaded = useSelector( getIsPartnerOAuthTokenLoaded );
```

Comment:
For retrieving products, we need to have the Partner OAuth token already loaded, if not, the API endpoint returns 403 Forbidden.

Reply:
This is the line causing one of the test failures. Changing 'enabled' to true or false instead of this value results in the test passing.
Test = `yarn run test-client client/jetpack-cloud/sections/partner-portal/primary/payment-method-add/test`.
The error itself is ` Error: Uncaught [TypeError: Cannot read properties of undefined (reading 'partner')]`.
I've not yet come up with a solution but flagging this now whilst I'm looking into it.

Reply:
Ok, if we `import 'calypso/state/partner-portal/init'`; then everything works. I'm not sure if this is recommended or ideal at this stage, just can confirm it works.

Reply:
Adding the following to `const initialState` into the only test in into `client/jetpack-cloud/sections/partner-portal/primary/payment-method-add/test/payment-method-add.jsx` does resolve the payment-method-add failing test in that file for me:

```
			partnerPortal: {
				partner: {
					current: {
						can_issue_licenses: true,
					},
				},
			},
```
(as does adding a partner with no additional values: `partner: {},`)

Somehow the similar would need to be applied in `clientstate/partner-portal/licenses/test/hooks.js`, but I've not worked out how to do that yet.

Reply:
I think we can 

- Add the below in `client/jetpack-cloud/sections/partner-portal/primary/payment-method-add/test/payment-method-add.jsx `

```
partnerPortal: {
				partner: {
					isPartnerOAuthTokenLoaded: true,
				},
			},
```

to the existing `initialState` like @coder-karen mentioned above

- Update the  useSelector as below in `client/state/partner-portal/licenses/test/hooks.js`

```
useSelector: jest.fn(),
```

Reply:
Great it works!